tokio/src/fs/file.rs

```diff
@@ -436,6 +436,27 @@ impl File {
             self.last_write_err = Some(e.kind());
         }
     }
+
+    /// Destructures the `tokio_fs::File` into a [`std::fs::File`][std].
+    ///
+    /// [std]: https://doc.rust-lang.org/std/fs/struct.File.html
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio::fs::File;
+    ///
+    /// # async fn dox() -> std::io::Result<()> {
+    /// let file = File::open("foo.txt").await?;
+    /// let std: std::fs::File = file.into_std();
+    ///
+    /// println!("{:?}", std);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn into_std(self) -> sys::File {
+        Arc::try_unwrap(self.std).unwrap()
```

Comment:
Ah... yeah, this can panic if there is an in-flight job working with the file. I'm not sure what to do about this.

Reply:
I'm not sure if it would be an alternative for a method, but I was thinking about adding a TryFrom implementation like this:

```rust
use std::convert::TryFrom;

impl TryFrom<File> for sys::File {
    type Error = File;

    fn try_from(f: File) -> Result<Self, Self::Error> {
        let File {
            std,
            state,
            last_write_err,
        } = f;
        Arc::try_unwrap(std).map_err(|std| File {
            std,
            state,
            last_write_err,
        })
    }
}
```

Reply:
We could do that, but the caller has no way to know if the file is at a point where it can succeed.

We could make it an async fn...

Reply:
I didn't really want to make this async, but we could just 'try_clone' the std file and return that.
'''
let std = self.std.clone();
asyncify(move || std.try_clone())
'''

Reply:
I think a `TryFrom` might work...